File: web-services/services/auditing/src/main/java/com/cryptocaddy/services/auditing/controller/UserController.java
package com.cryptocaddy.services.auditing.controller;

import com.cryptocaddy.services.common.user.ApplicationUser;
import com.cryptocaddy.services.common.user.ApplicationUserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/users")
public class UserController {

    //private ApplicationUserRepository applicationUserRepository;
    private BCryptPasswordEncoder bCryptPasswordEncoder;


    @Autowired
    public UserController(BCryptPasswordEncoder bCryptPasswordEncoder) {
        //this.applicationUserRepository = applicationUserRepository;
        this.bCryptPasswordEncoder = bCryptPasswordEncoder;
    }

//    public UserController(BCryptPasswordEncoder bCryptPasswordEncoder) {
//        this.bCryptPasswordEncoder = bCryptPasswordEncoder;
//    }

    @PostMapping("/sign-up")
    public void signUp(@RequestBody ApplicationUser user) {
        user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
        //applicationUserRepository.save(user);
    }

}

File: web-services/services/auditing/src/main/java/com/cryptocaddy/services/auditing/api/UserController.kt
package com.cryptocaddy.services.auditing.api

import com.cryptocaddy.libraries.database.dao.User
import com.cryptocaddy.libraries.database.dao.UserRepository
import com.cryptocaddy.services.auditing.dto.response.Result
import com.cryptocaddy.services.auditing.dto.request.RequestCreateAccount
import com.cryptocaddy.services.common.authentication.JWTAuthenticator
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*


@RestController
@RequestMapping("/users")
class UserController(val userRepository: UserRepository) {

    @GetMapping(value = ["/{id}"], produces = ["application/json"])
    fun getUser(@PathVariable("id") id: String): User? =
            userRepository.findOne(id)


    @PostMapping(value = ["/add"], produces = ["application/json"], consumes = ["application/json"])
    fun addUser(@RequestBody requestCreateAccount: RequestCreateAccount): ResponseEntity<Result?> {

        val jwtBody = JWTAuthenticator.getBodyFromToken(requestCreateAccount.token)
        val user = User(jwtBody.uid, jwtBody.email, jwtBody.name)
        userRepository.save(user)

        //TODO: more error handling
        val response = Result("Success")
        return ResponseEntity(response, HttpStatus.OK)

    }

}

